integration-tests/test_registration.py

```diff
@@ -0,0 +1,56 @@
+import os
+import subprocess
+
+import pytest
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+MACHINE_ID_FILE: str = "/etc/insights-client/machine-id"
+
+
+def test_machineid_exists_only_when_registered(insights_client):
+    """`machine-id` is only present when insights-client is registered."""
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    res = insights_client.run(check=False)
+    assert (
+        "This host has not been registered. Use --register to register this host."
+    ) in res.stdout
+    assert res.returncode != 0
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.register()
+    assert os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.unregister()
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+
+@pytest.mark.skip("This has always been failing")
```

Comment:
what is exactly failing always? Is it the 'register' operation second time? I see the expected message when registering twice"This host has already been registered" and exit code 0. This is the so far expected behavior and no bugs logged for exit code.

Reply:
Hi, this test failed in the manual verification, you can take a look into the documentation of cct-161 manual verification on the version of insights:) that's why (I think) Matyas said that that's still failing as we both tried it and failed it for both of us while manually verifying it.

Reply:
`RHELPLAN-117674` should have more context. Maybe we should mention that somehow.

```py
@pytest.mark.skipped("Broken use-case: RHELPLAN-117674")
```

Reply:
In my opinion I think we are confusing two different test cases here, the issue/bug you highlighted is applicable if insights is unregistered and machine-id exist

Reply:
or rather using `pytest.mark.xfail()`, possibly with an explicit `strict=False`: this way, in case this is fixed in insights-core, old versions/tags of insights-client not will fail (and only get a non-fatal XPASS)